mesonbuild/compilers/cpp.py

```diff
@@ -447,6 +448,11 @@ def get_option_compile_args(self, options: 'KeyedOptionDictType') -> T.List[str]
         if not options[key.evolve('rtti')].value:
             args.append('-fno-rtti')
 
+        # XXX: This needs updating if/when GCC starts to support libc++.
+        # It currently only does so via an experimental configure arg.
+        if not are_asserts_disabled(options):
+            args.append('-D_GLIBCXX_ASSERTIONS=1')
```

Comment:
Maybe this makes more sense in `get_assert_args(disable: bool)`? (That is where `-DNDEBUG` is currently handled.)